challenge-sizebay/src/Components/ProgressBar/ProgressBar.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState, useContext } from "react";
+import { useLayoutEffect } from "react";
+import { GlobalContext } from "../../Contexts/GlobalContext";
+import { ProgressWrapper, ProgressDone } from "./style";
+import confetti from 'canvas-confetti'
+import { useEffect } from "react";
+
+const ProgressBar = () => {
+  const [style, setStyle] = useState({});
+  const { states } = useContext(GlobalContext);
+
+  const percentageDone = (part, total) => part * 100 / total;
```

Comment:
ref: this can be improved by renaming the method to `getPercentageDone` and wrapped around a `useMemo` to avoid extra memory performs.

Reply:
Hey there! I've just implemented it, but again, I need to get a better idea of when it would be ideal to use.

Reply:
`useCallback` is usually preferred when you're requesting the browser to calculate expensive layout changes or formats. That said: it allows you to cache a function definition between re-renders, to avoid unnecessary memory leaks.

You declare pretty much a normal function that returns a function as a callback, like this:
```js
// before
const percentageDone = (part, total) => part * 100 / total;

// after
const percentageDone = useCallback((part, total) => part * 100 / total, [])
```

`useMemo` is a bit different because, instead of a function that wraps a function, it's a variable that wraps a function. It's preferrable to adopt `useCallback` on this one because you're passing two arguments to compose your formulae.